src/python_testing/drlk_2_x_common.py

```diff
@@ -0,0 +1,309 @@
+                                                                                              credential=credential,
+                                                                                              credentialData=credentialData,
+                                                                                              userIndex=userIndex,
+                                                                                              userStatus=userStatus,
+                                                                                              userType=userType),
+                                         endpoint=self.endpoint,
+                                         timedRequestTimeoutMs=1000)
+        asserts.assert_true(type_matches(ret, Clusters.Objects.DoorLock.Commands.SetCredentialResponse),
+                            "Unexpected return type for SetCredential")
+        asserts.assert_true(ret.status == Status.Success, "Error sending SetCredential command, status={}".format(str(ret.status)))
+        return ret
+
+    async def send_clear_credential_cmd(self, credential) -> None:
+        await self.send_single_cmd(cmd=Clusters.Objects.DoorLock.Commands.ClearCredential(credential=credential),
+                                   endpoint=self.endpoint,
+                                   timedRequestTimeoutMs=1000)
+        ret = await self.send_single_cmd(cmd=Clusters.Objects.DoorLock.Commands.GetCredentialStatus(credential=self.createdCredential),
+                                         endpoint=self.endpoint)
+        asserts.assert_true(type_matches(ret, Clusters.Objects.DoorLock.Commands.GetCredentialStatusResponse),
+                            "Unexpected return type for GetCredentialStatus")
+        asserts.assert_false(ret.credentialExists, "Error clearing Credential (credentialExists==True)")
+
+    async def cleanup_users_and_credentials(self):
+        self.print_step("Cleanup", "Clear created User and Credential on the DUT")
+        if self.createdCredential:
+            await self.send_clear_credential_cmd(self.createdCredential)
+            logging.info("Credential cleared at CredentialIndex %d" % (self.createdCredential.credentialIndex))
+        self.createdCredential = None
+
+    async def generate_pincode(self, maxPincodeLength):
+        return ''.join(random.choices(string.digits, k=maxPincodeLength))
+
+    async def teardown(self):
+        await self.cleanup_users_and_credentials()
+
+    async def run_drlk_test_common(self, lockUnlockCommand, lockUnlockCmdRspPICS, lockUnlockText, doAutoRelockTest):
+        is_ci = self.check_pics('PICS_SDK_CI_ONLY')
+
+        self.createdCredential = None
+
+        self.endpoint = self.user_params.get("endpoint", 1)
+
+        # Allow for user overrides of these values
+        credentialIndex = self.user_params.get("credential_index", 1)
```

Comment:
Are these values (all the ones from the user params) exposed through device attributes?

If not...should they be? I suppose that's a spec question. If they are, they should probably be read from the device.

Reply:
Values such as endpoint could (in the future) be dynamically determined by walking the descriptor lists. These are here currently to allow for a TC run with non-standard values (e.g., a multi-function DUT with the DoorLock cluster on another EP)